.gitignore

```diff
@@ -359,4 +359,9 @@ build/
 # Default Assets restore directory
 .assets
 
-CMakeUserPresets.json
+CMakeUserPresets.json
+
+# Disable VS solution files and csproj files because they pollute the VS experience.
+**/vendor/**/*.sln
+**/vendor/**/*.csproj
```

Comment:
I haven't tried this locally, maybe you have, will VS ignore files that are ignored by git like this?

It doesn't ignore the `.github` folder, when opening the repo directory in VS.

Reply:
I tried this on my DevBox. Before the change, the sln file in the vendor directory was selected as the default (and when I added just the sln file, it chose a .csproj file), afterwards it was not.

Reply:
Because I wanted to confirm that what I saw yesterday was not my imagination, I went to my DevBox instance, deleted the two lines from the .gitignore file, and deleted the contents of the `.vs` directory.

I then launched visual studio and the .sln file was selected as the default target.

I exited visual studio, removed the `.vs` directory, added the two lines to the .gitignore file, and restarted visual studio.

The root directory was selected in the solution view and "Select a project" was the default target.